apps/io-wallet-user-func/src/infra/jwt-validator.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { HubSpidLoginConfig } from "@/app/config";
+import { UnauthorizedError } from "@/error";
+import { ExchangeJwtValidate, HslJwtValidate } from "@/jwt-validator";
+import { getValidateJWT } from "@pagopa/ts-commons/lib/jwt_with_key_rotation";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, pipe } from "fp-ts/lib/function";
+import * as t from "io-ts";
+import * as jwt from "jsonwebtoken";
+
+const validateAndDecode: (
+  issuer: NonEmptyString,
+  pubKey: NonEmptyString,
+) => (token: NonEmptyString) => TE.TaskEither<Error, jwt.JwtPayload> =
+  (issuer, pubKey) => (token) =>
+    pipe(token, getValidateJWT(issuer, pubKey));
+
+const IntrospectSuccessResponse = t.type({
+  active: t.boolean,
+});
+
+const hslIntrospection: (
+  clientBaseUrl: NonEmptyString,
+) => (token: NonEmptyString) => TE.TaskEither<Error, void> =
+  (clientBaseUrl) => (token) =>
+    pipe(
+      TE.tryCatch(async () => {
+        const result = await fetch(new URL("/introspect", clientBaseUrl), {
+          body: JSON.stringify({
+            token,
+          }),
+          method: "POST",
+          signal: AbortSignal.timeout(3000),
```

Comment:
How about making this timeout configurable? Maybe with a default value if not present in ENV. Just like above where the todo was

Reply:
I've added a iira task so we can think about it properly, since that the timeout is present in all HTTP calls